src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -350,14 +350,13 @@ def protect_subvolume_snapshot(self, **kwargs):
         ret        = 0, "", ""
         volname    = kwargs['vol_name']
         subvolname = kwargs['sub_name']
-        snapname   = kwargs['snap_name']
         groupname  = kwargs['group_name']
 
         try:
             with open_volume(self, volname) as fs_handle:
                 with open_group(fs_handle, self.volspec, groupname) as group:
                     with open_subvol(fs_handle, self.volspec, group, subvolname) as subvolume:
-                        subvolume.protect_snapshot(snapname)
+                        log.warning("snapshot protect call is deprecated and will be removed in a future release")
```

Comment:
This is meeting errors half-way, we do check if subvolume exists but not if the snap exists as this is pretty much a no-op.

I felt fine doing this, but other options are,
- Check for snapshot existence, and then do nothing
- Forgo the check for the subvolume, group and volume, and return success from `_cmd_fs_subvolume_snapshot_protect`

Reply:
Makes sense; this lgtm, thanks!

Reply:
Looks good, thanks.

Reply:
Looks good to me too, I don't see too much value to the desired output on adding/removing checks :)